apps/src/blockly/addons/cdoFieldImageDropdown.ts

```diff
@@ -127,6 +134,82 @@ export class CdoFieldImageDropdown extends FieldGridDropdown {
       });
     }
   }
+
+  /**
+   * Sets the field's state based on the given state value. Should only be
+   * called by the serialization system.
+   *
+   * @param state The state we want to apply to the field.
+   * @override because `state` is stored as either json or xml in our code base.
+   */
+  // Blockly uses any here
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  loadState(state: any) {
```

Comment:
The methods `loadState`, `fromXml`, and `toXml` are standard Blockly serialization hooks. The code for each was copied directly from `CdoFieldDropdown`. That class and this one both extended Core Blockly classes so each needs to have its own method overrides to support our custom serialization needs. Specifically, those needs are around the `config` XML attribute which allows to control which field values are available per block instance.